src/main/java/io/openraven/magpie/api/MagpieResource.java

```diff
@@ -210,4 +218,118 @@ public JsonNode getDiscoveryMeta() {
   public void setDiscoveryMeta(JsonNode discoveryMeta) {
     this.discoveryMeta = discoveryMeta;
   }
+
+  public static class MagpieResourceBuilder {
+    private final ObjectMapper mapper;
+    private String assetId;
+    private String resourceName;
+    private String resourceId;
+    private String resourceType;
+    private String region;
+    private String projectId;
+    private String accountId;
+    private Instant createdIso;
+    private Instant updatedIso = Instant.now();
+    private String discoverySessionId;
+    private Long maxSizeInBytes = null;
+    private Long sizeInBytes = null;
+    private JsonNode configuration;
+    private JsonNode supplementaryConfiguration;
+    private JsonNode tags;
+    private JsonNode discoveryMeta;
+
+    public MagpieResourceBuilder(ObjectMapper mapper, String assetId) {
```

Comment:
Would it be better if MagpieResourceBuilder would have its own ObjectMapper instead of taking one as arugment?

Reply:
I would keep MagpieResource as generic as possible as I see that mapper is created differently for AWS and GCP. So might be better to leave mapper creation to plugin